validation-test/compiler_crashers_2/0083-rdar31163470-2.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: not --crash %target-swift-frontend -primary-file %s -emit-ir
+// REQUIRES: asserts
+
+protocol C {
+  associatedtype I
+}
+
+protocol PST {
+  associatedtype LT : C
+}
+
+protocol SL {
+  associatedtype S : PST
+}
+
+struct PEN<_S : PST> : SL {
+  typealias S = _S
+  let l: _S.LT.I
+}
+
+struct PE<N : SL> {
+  let n: N
+  static func c<S>(_: PE<N>) where N == PEN<S> {}
```

Comment:
The minimized signature of PE.c becomes `<N, S where N == PEN<S>>`; we've lost the conformance requirement `S : PST`. If I add it explicitly, the GenericSignatureBuilder warns that it is redundant. This feels wrong, because while it is inferred by the application of the type constructor `PEN` to `S`, it should still appear in the minimized signature.